flow/Platform.actor.cpp

```diff
@@ -3497,35 +3497,57 @@ extern void initProfiling();
 std::atomic<double> checkThreadTime;
 #endif
 
-volatile thread_local bool profileThread = false;
-volatile thread_local int profilingEnabled = 1;
+// True if this thread is the thread being profiled. Not to be used from the signal handler.
+thread_local bool profileThread = false;
 
-volatile thread_local int64_t numProfilesDeferred = 0;
-volatile thread_local int64_t numProfilesOverflowed = 0;
-volatile thread_local int64_t numProfilesCaptured = 0;
-volatile thread_local bool profileRequested = false;
+// The thread ID of the profiled thread. This can be compared against the current thread ID
+// to see if we are on the profiled thread. Can be used in the signal handler.
+volatile int64_t profileThreadId = -1;
```

Comment:
On the client-side there can be multiple network threads, so the profiling would work only on one of them. If we need to support multiple thread, we should probably think about some thread-safe data structure to replace thread-local state.

Reply:
It's not a current requirement that we support multiple threads, at least in a single client library. It's possible that could change at some point in the future if we allow running multiple threads, but I suspect that would complicate many things about this implementation and can be deferred until we need it.